core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -308,6 +308,11 @@ public static boolean isConflictException(Throwable t) {
         ShardRouting routingEntry();
 
         /**
+         * primary term for this primary (fixed when primary shard is created)
+         */
+        long primaryTerm();
```

Comment:
this also means that we can fold setting the primary term on a request into the ReplicationOperation - we currently only assert that it was set.